python/cudf/dataframe/dataframe.py

```diff
@@ -453,18 +453,21 @@ def take(self, positions, ignore_index=False):
             out[col] = self[col].take(positions, ignore_index=ignore_index)
         return out
 
-    def copy(self):
+    def copy(self, deep=True):
         """
         Returns a copy of this dataframe
         """
         df = DataFrame()
         df._index = self._index
         df._size = self._size
-        df._cols = self._cols.copy()
+        if deep:
+            df._cols = self._cols.copy()
+        else:
+            df._cols = self._cols
```

Comment:
This will pass the OrderedDict by reference so changes to the resulting DataFrame will affect the calling DataFrame. We need a new OrderedDict and pass the columns by reference in the case of a shallow copy.